### ISSUE TITLE
Give ability to customise Wrapper and/or WrapperArray

### ISSUE THREAD

**Issue by davidlandais** [vuejs/vue-test-utils#1224]
### What problem does this feature solve?
I would like to propose you to give the ability to developers to register custom commands like wrapper.`findElementWithTestIdAttribute()`.isVisible() or wrapper.`findVueAndEmit(selector, ['event', ...args])`. This is my needs, that i work around with some helpers. This is for example, a feature shipped with Cypress. What do you think ? Does it sound a good idea ?

### What does the proposed API look like?
The API doesn't seems to propose this feature.

**Comment by winniehell** [comment]
In https://github.com/vuejs/vue-test-utils/pull/1245#discussion_r289268663, @souldzin suggested using an `EventEmitter` to hook into `Wrapper` creation. A Vue instance as event hub (described [here](https://vuejs.org/v2/guide/migration.html#dispatch-and-broadcast-replaced)) would serve the same purpose.

We could then do the following:

```javascript
import { events } from '@vue/test-utils';

events.$on('wrapper-created', newWrapper => {
  Object.assign(newWrapper. {
    findElementWithTestIdAttribute() {
      // ...
    },

    isVisible() {
      // ...
    },

    findVueAndEmit() {
      // ...
    },
  });
});
```

or alternatively:

```javascript
import { Wrapper } from '@vue/test-utils';

Wrapper.events.$on('created', newWrapper => {
  // ...
});
```

@eddyerburgh What do you think?

**Comment by winniehell** [comment]
@eddyerburgh If the above approach makes sense to you, I would be happy to come up with a pull request to unblock https://github.com/vuejs/vue-test-utils/issues/1236.

**Comment by winniehell** [comment]
@JessicaSachs Can you please take a look at this?

**Comment by JessicaSachs** [comment]
@winniehell for back and forth convo can you find me on Vue Land discord? It’s easier to chat there.

**Comment by winniehell** [comment]
@JessicaSachs I didn't have the impression that there was much back and forth yet and also would like to have the result documented here. :sweat_smile: But sure, we can explore the different options on discord first. :+1:

**Comment by JessicaSachs** [comment]
Agreed and agreed, but I want to talk it over with you and then we will
report back to GH so that other maintainers can weigh in.

**Comment by JessicaSachs** [comment]
Implementing this feature request would make it easier to fix a bug (#1236) by way of an existing PR (#1245) that wants to depend on this new feature.

@winniehell and I will look into ways of solving #1236 without depending on this feature request.

**Comment by davidlandais** [comment]
Ok but the main idea is to share those "commands" between wrappers. For example, if I have a login-view.spec.js and a register-view.spec.js files which are both using a form. For example, I would like to fill form with ease. I could make a "command" called`fillForm` which could find X inputs and `setValue` to each one. like `loginViewWrapper.fillForm('#login', { email: 'j.doe@unkown.lost' })`. Do you think we could share this kind of "helpers" between spec files easily ? Regarding your ideas, the "setupFilesAfterEnv" Jest config could be a good place. What do you think ?

**Comment by JessicaSachs** [comment]
@davidlandais Right now the API is frozen because we’re trying to get v1 out so we’re only doing bug fixes. I didn’t mean anything about the feature request by saying we weren’t going to pick it up right now.
